src/basic/cgroup-util.c

```diff
@@ -2635,20 +2653,33 @@ int cg_enable_everywhere(CGroupMask supported, CGroupMask mask, const char *p) {
 
                         if (!f) {
                                 f = fopen(fs, "we");
-                                if (!f) {
-                                        log_debug_errno(errno, "Failed to open cgroup.subtree_control file of %s: %m", p);
-                                        break;
-                                }
+                                if (!f)
+                                        return log_debug_errno(errno, "Failed to open cgroup.subtree_control file of %s: %m", p);
                         }
 
                         r = write_string_stream(f, s, WRITE_STRING_FILE_DISABLE_BUFFER);
                         if (r < 0) {
-                                log_debug_errno(r, "Failed to enable controller %s for %s (%s): %m", n, p, fs);
+                                log_debug_errno(r, "Failed to %s controller %s for %s (%s): %m",
+                                                FLAGS_SET(mask, bit) ? "enable" : "disable", n, p, fs);
                                 clearerr(f);
+
+                                /* If we can't turn off a controller, leave it on in the reported resulting mask. This
+                                 * happens for example when we attempt to turn off a controller up in the tree that is
+                                 * used down in the tree. */
+                                if (!FLAGS_SET(mask, bit) && r == -EBUSY)
```

Comment:
Hmm, isn't this enough without the `EBUSY` check? Not sure it buys us much here

Reply:
so, when disabling a controller EBUSY indicates "controller is enabled, but we can't disable it because something further down the tree still uses it". An error like EINVAL or EOPNOTSUPP or so very likely means "i never heard of this controller, what are you talking about?". In the former case this means we can assume a-posteriori that the controller continues to be enabled, in the latter case we cannot assume that, it probably doesn't exist at all..

i will also add a comment about that